install.sh

```diff
@@ -0,0 +1,437 @@
+    err "unrecognized OS type: $_ostype"
+    ;;
+  esac
+
+  case "$_cputype" in
+  i386 | i486 | i686 | i786 | x86)
+    _cputype=i686
+    ;;
+
+  xscale | arm)
+    _cputype=arm
+    if [ "$_ostype" = "linux-android" ]; then
+      _ostype=linux-androideabi
+    fi
+    ;;
+
+  armv6l)
+    _cputype=arm
+    if [ "$_ostype" = "linux-android" ]; then
+      _ostype=linux-androideabi
+    else
+      _ostype="${_ostype}eabihf"
+    fi
+    ;;
+
+  armv7l | armv8l)
+    _cputype=armv7
+    if [ "$_ostype" = "linux-android" ]; then
+      _ostype=linux-androideabi
+    else
+      _ostype="${_ostype}eabihf"
+    fi
+    ;;
+
+  aarch64 | arm64)
+    _cputype=aarch64
+    ;;
+
+  x86_64 | x86-64 | x64 | amd64)
+    _cputype=x86_64
+    ;;
+
+  mips)
+    _cputype=$(get_endianness mips '' el)
+    ;;
+
+  mips64)
+    if [ "$_bitness" -eq 64 ]; then
+      # only n64 ABI is supported for now
+      _ostype="${_ostype}abi64"
+      _cputype=$(get_endianness mips64 '' el)
+    fi
+    ;;
+
+  ppc)
+    _cputype=powerpc
+    ;;
+
+  ppc64)
+    _cputype=powerpc64
+    ;;
+
+  ppc64le)
+    _cputype=powerpc64le
+    ;;
+
+  s390x)
+    _cputype=s390x
+    ;;
+  riscv64)
+    _cputype=riscv64gc
+    ;;
+  loongarch64)
+    _cputype=loongarch64
+    ;;
+  *)
+    err "unknown CPU type: $_cputype"
+    ;;
+  esac
+
+  # Detect 64-bit linux with 32-bit userland
+  if [ "${_ostype}" = unknown-linux-gnu ] && [ "${_bitness}" -eq 32 ]; then
+    case $_cputype in
+    x86_64)
+      if [ -n "${RUSTUP_CPUTYPE:-}" ]; then
+        _cputype="$RUSTUP_CPUTYPE"
+      else {
+        # 32-bit executable for amd64 = x32
+        if is_host_amd64_elf; then
+          err "x86_64 linux with x86 userland unsupported"
+        else
+          _cputype=i686
+        fi
+      }; fi
+      ;;
+    mips64)
+      _cputype=$(get_endianness mips '' el)
+      ;;
+    powerpc64)
+      _cputype=powerpc
+      ;;
+    aarch64)
+      _cputype=armv7
+      if [ "$_ostype" = "linux-android" ]; then
+        _ostype=linux-androideabi
+      else
+        _ostype="${_ostype}eabihf"
+      fi
+      ;;
+    riscv64gc)
+      err "riscv64 with 32-bit userland unsupported"
+      ;;
+    esac
+  fi
+
+  # Detect armv7 but without the CPU features Rust needs in that build,
+  # and fall back to arm.
+  # See https://github.com/rust-lang/rustup.rs/issues/587.
+  if [ "$_ostype" = "unknown-linux-gnueabihf" ] && [ "$_cputype" = armv7 ]; then
+    if ensure grep '^Features' /proc/cpuinfo | grep -q -v neon; then
+      # At least one processor does not have NEON.
+      _cputype=arm
+    fi
+  fi
+
+  _arch="${_cputype}-${_ostype}"
+
+  RETVAL="$_arch"
+}
+
+resolve_version() {
+  local _requested_version=$1
+  local _requested_ref=$2
+
+  local _response
+
+  say "retrieving $_requested_version version from ${SCARB_REPO}..."
+  _response=$(ensure curl -fLs -H 'Accept: application/json' "${SCARB_REPO}/releases/${_requested_ref}")
+  if [ "$_response" == "{\"error\":\"Not Found\"}" ]; then
```

Comment:
It looks like (again, dash) has some problems with it, it looks like it does not support '==' operator

```
$ echo $0
dash
$ response="{\"error\":\"Not Found\"}"
$ if [ "$_response" == "{\"error\":\"Not Found\"}" ]; then echo "noanie"; else echo "nowim"; fi
dash: 9: [: {"error":"Not Found"}: unexpected operator
nowim
$ if [ "$_response" = "{\"error\":\"Not Found\"}" ]; then echo "noanie"; else echo "nowim"; fi
noanie
$
```

Reply:
Fixed